tests/test_xpaths.py

```diff
@@ -103,22 +128,15 @@ def setUp(self):
             )
 
     def test_partners(self):
-        response = self.reply.as_dict()
-        persoon = response['persoon']
+        response = self.reply.as_json()
+        data = json.loads(response)
 
-        # Test data needs to be improved to test the merging.
-        self.assertEqual(persoon['heeftAlsEchtgenootPartner'][0]['gerelateerde']['bsn'], '456789013')
-        self.assertEqual(persoon['heeftAlsEchtgenootPartner'][1]['gerelateerde']['bsn'], '456789014')
-
-        self.assertEqual(persoon['heeftAlsEchtgenootPartner'][0]['datumSluiting'], '1974-01-01T00:00:00')
-        self.assertFalse(persoon['heeftAlsEchtgenootPartner'][0].get('datumOntbinding', False))
-        # test data does not have datumSluiting for ex-partner
-        self.assertEqual(persoon['heeftAlsEchtgenootPartner'][1]['datumOntbinding'], '1973-01-01T00:00:00')
+        # deep equal test
+        self.assertEqual(data['verbintenis'], self.get_result())
```

Comment:
Personally I like to see as much of the test where it is actually tested. I don't see why using a function here makes the test any easier to maintain or comprehend.

Reply:
Personally i find it quite unreadable. (also hard to balance with flake8)